ocs_ci/deployment/assisted_installer.py

```diff
@@ -0,0 +1,396 @@
+                    if v["status"] in ("failure", "pending"):
+                        failed_validations.append(
+                            f"host {host['id']}, section {section}, {v['id']}: {v['status']} ({v['message']})"
+                        )
+        if failed_validations:
+            msg = f"Failed hosts validations: \n{failed_validations.join(os.linesep)}"
+            logger.error(msg)
+            raise HostValidationFailed(msg)
+        logger.info("Host validations passed on all hosts.")
+
+    def get_host_id_mac_mapping(self):
+        """
+        Prepare mapping between host ID and mac addresses
+
+        Return:
+            dict: host id to mac mapping
+        """
+        hosts = self.api.get_infra_env_hosts(self.infra_id)
+        return {
+            h["id"]: json.loads(h["inventory"])["interfaces"][0]["mac_address"]
+            for h in hosts
+        }
+
+    def update_hosts_config(self, mac_name_mapping, mac_role_mapping):
+        """
+        Update host names and roles.
+
+        Args:
+            mac_name_mapping (dict): host mac address to host name mapping
+            mac_role_mapping (dict): host mac address to host role mapping
+        """
+        host_id_mac_mapping = self.get_host_id_mac_mapping()
+        for host_id in host_id_mac_mapping:
+            update_data = {
+                "host_name": mac_name_mapping[host_id_mac_mapping[host_id]],
+                "host_role": mac_role_mapping[host_id_mac_mapping[host_id]],
+            }
+            self.api.update_infra_env_host(self.infra_id, host_id, update_data)
+            logger.info(f"Updated host {host_id} configuration: {update_data}")
+
+    def install_cluster(self):
+        """
+        Trigger cluster installation
+        """
+        self.api.install_cluster(self.id)
+        logger.info("Started cluster installation")
+        # wait for cluster installation success
+        for sample in TimeoutSampler(
+            timeout=3600, sleep=300, func=self.api.get_cluster, cluster_id=self.id
+        ):
+            # TODO: add more information about cluster installation progress
+            logger.info(
+                f"Cluster installation status: {sample['status']} ({sample['status_info']})"
+            )
+            if sample["status"] == "installed":
+                logger.info(
+                    f"Cluster was successfully installed (status: {sample['status']} - {sample['status_info']})"
+                )
+                break
+        # create metadata, kubeconfig, kubeadmin-password and openshift_install.log files
+        self.create_metadata_file()
+        self.create_kubeconfig_file()
+        self.create_kubeadmin_password_file()
+        self.create_openshift_install_log_file()
+
+    def create_metadata_file(self):
+        """
+        Create .openshift_install.log file containing URL to OpenShift console
+        It is used by our CI jobs to show the console URL in build description.
```

Comment:
probably description here miscopied from the next func create_openshift_install_log_file